### PR TITLE
feat(compiler): allow setting attributes on a host element

### LINKED ISSUE #1402: Create hostAttributes support
## Goal

To be able to statically describe attributes on host element.

Example: <md-button> needs to be declared as <md-button aria-role="button"> for proper behavior. Doing this imperatively would be unnecessarily slow. Changing the template is more effective.

## API

```javascript
@Component({
  selector: 'md-button',
  hostAttributes: {
    'role': 'button'
  }
})
@View(...)
class MdButton {}
```

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Closes #1402

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@mhevery @tbosch here is a first stab at `hostAttributes` support. I'm still wrapping my head around the entire compiler after the render layer split so I'm not sure if we can simply sent attributes on a host element during compilation or maybe we should hope through some abstractions here.

The other corner case which I'm not sure how to tackle is when there are multiple directives trying to set the same attribute with different values. For now "first one wins" but maybe we should throw?

File: modules/angular2/test/render/dom/compiler/directive_parser_spec.js [hunk 1]
```diff
@@ -123,6 +125,20 @@ export function main() {
...
+      var results = process(element);
+
+      expect(DOM.getAttribute(results[0].element, 'attr_name')).toEqual('attr_val');
+    });
+
+    it('should not set host element attribute if an attribute already exists', () => {
+      var element = el('<input attr_name="initial" some-decor-with-host-attrs>');
+      var results = process(element);
+
+      expect(DOM.getAttribute(results[0].element, 'attr_name')).toEqual('initial');
```
Comment (mhevery) [inline modules/angular2/test/render/dom/compiler/directive_parser_spec.js]: if the directive renames the attribute this test would still pass. I think this test should add removal of attribute and call `process` one more time to show that now the attribute gets added.

Comment (pkozlowski-opensource) [inline modules/angular2/test/render/dom/compiler/directive_parser_spec.js]: @mhevery 
> if the directive renames the attribute

Hmm, I'm not sure how a directive could rename / remove an attribute :-/ Will ping you on chat to better understand the use-case that I'm missing here.

Comment (mhevery) [inline modules/angular2/test/render/dom/compiler/directive_parser_spec.js]: Sorry for the confusion.

If the author of the Directive changes the definition and renames `attr_name` to `my_attr_name` then this test would still pass.

**Pull request merged**